test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -4431,6 +4431,70 @@ public async Task Purge_Single_Instance_History(bool extendedSessions, string st
             }
         }
 
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [MemberData(nameof(TestDataGenerator.GetBooleanAndFullFeaturedStorageProviderOptions), MemberType = typeof(TestDataGenerator))]
+        public async Task Purgemultipleinstancehistory(bool extendedSessions, string storageProvider)
+        {
+            using (ITestHost host = TestHelpers.GetJobHost(
+               this.loggerProvider,
+               nameof(this.Purgemultipleinstancehistory),
+               extendedSessions,
+               storageProviderType: storageProvider))
+            {
+                await host.StartAsync();
+
+                var client1 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "foo", this.output);
+                var client2 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "bar", this.output);
+                var client3 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "baz", this.output);
+
+                string firstInstanceId = client1.InstanceId;
+                string secondInstanceId = client2.InstanceId;
+                string thirdInstanceId = client3.InstanceId;
+                IList<string> instanceID = new List<string> { firstInstanceId, secondInstanceId, thirdInstanceId };

```

Comment:
For variables that contain a list of values, it's best to make the variable name plural or add a suffix to make it clear that the variable contains multiple values. For example, `instanceIds` or `instanceIdList` would be better names.